src/main/java/de/tum/in/www1/artemis/repository/StatisticsRepository.java

```diff
@@ -114,4 +205,343 @@
+            listElement.put("day", start);
+            listElement.put("amount", (long) v.size());
+            returnList.add(listElement);
+        });
+        return returnList;
+    }
+
+    private Integer getMonthIndex(Month month) {
+        return switch (month) {
+            case JANUARY -> 1;
+            case FEBRUARY -> 2;
+            case MARCH -> 3;
+            case APRIL -> 4;
+            case MAY -> 5;
+            case JUNE -> 6;
+            case JULY -> 7;
+            case AUGUST -> 8;
+            case SEPTEMBER -> 9;
+            case OCTOBER -> 10;
+            case NOVEMBER -> 11;
+            case DECEMBER -> 12;
+        };
+    }
+
+    default Integer getWeekOfDate(ZonedDateTime date) {
+        LocalDate localDate = date.toLocalDate();
+        TemporalField woy = WeekFields.of(DayOfWeek.MONDAY, 4).weekOfWeekBasedYear();
+        return localDate.get(woy);
+    }
+
+    /**
+     * Gets a list of maps, each map describing an entry in the database. The map has the two keys "day" and "amount",
+     * which map to the date and the amount of the findings. This method handles the spanType DAY
+     *
+     * @param outcome A List<Map<String, Object>>, containing the content which should be refactored into an array
+     * @param result the array in which the converted outcome should be inserted
+     * @param endDate the endDate
+     * @return an array, containing the values for each bar in the graph
+     */
+    default Integer[] createResultArrayForDay(List<Map<String, Object>> outcome, Integer[] result, ZonedDateTime endDate) {
+        for (Map<String, Object> map : outcome) {
+            int hour = ((ZonedDateTime) map.get("day")).getHour();
+            Integer amount = map.get("amount") != null ? ((Long) map.get("amount")).intValue() : 0;
+            for (int i = 0; i < 24; i++) {
+                if (hour == endDate.minusHours(i).getHour()) {
+                    result[endDate.getHour() - i] += amount;
+                }
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Gets a list of maps, each map describing an entry in the database. The map has the two keys "day" and "amount",
+     * which map to the date and the amount of the findings. This method handles the spanType WEEK
+     *
+     * @param outcome A List<Map<String, Object>>, containing the content which should be refactored into an array
+     * @param result the array in which the converted outcome should be inserted
+     * @param endDate the endDate
+     * @return an array, containing the values for each bar in the graph
+     */
+    default Integer[] createResultArrayForWeek(List<Map<String, Object>> outcome, Integer[] result, ZonedDateTime endDate) {
+        for (Map<String, Object> map : outcome) {
+            ZonedDateTime date = (ZonedDateTime) map.get("day");
+            Integer amount = map.get("amount") != null ? ((Long) map.get("amount")).intValue() : 0;
+            for (int i = 0; i < 7; i++) {
```

Comment:
I wonder whether there is no other mechanism to aggregate the data by day

Reply:
Improved the mechanism so we do not have to iterate through the possible spots